junit-runtime/src/main/scala/org/scalajs/junit/JUnitTask.scala

```diff
@@ -143,6 +127,30 @@ private[junit] final class JUnitTask(val taskDef: TaskDef,
     }
   }
 
+  private def reportExecutionErrors(reporter: Reporter, method: Option[String],
+      timeInSeconds: Double, errors: List[Throwable]): Int = {
+    errors match {
+      case Nil =>
+        // fast path
+        0
+
+      case e :: Nil if isAssumptionViolation(e) =>
+        reporter.reportAssumptionViolation(method, timeInSeconds, e)
+        0
+
+      case _ =>
+        val errorsPatchedForAssumptionViolations = errors.map {
+          case error: org.junit.internal.AssumptionViolatedException =>
```

Comment:
Are you sure this is the right behavior and not `isAssumptionViolation`? (or maybe that one is wrong) Just seems odd that they are different.

Reply:
It's what JUnit does, at least:
https://github.com/junit-team/junit4/blob/9ad61c6bf757be8d8968fd5977ab3ae15b0c5aba/src/main/java/org/junit/runners/model/MultipleFailureException.java#L35-L37
```java
            if (error instanceof AssumptionViolatedException) {
                error = new TestCouldNotBeSkippedException((AssumptionViolatedException) error);
            }
```
with the import at
https://github.com/junit-team/junit4/blob/9ad61c6bf757be8d8968fd5977ab3ae15b0c5aba/src/main/java/org/junit/runners/model/MultipleFailureException.java#L10
```java
import org.junit.internal.AssumptionViolatedException;
```
I think `isAssumptionViolation` is doing more work than necessary. `org.junit.AssumptionViolatedException` *extends* `org.junit.internal.AssumptionViolatedException`. That doesn't seem related to this upgrade, though. Should I add a commit to simplify it?

Reply:
Well, I merged the simplification in this commit, because it made more sense, actually.